AustinHarris.JsonRpcTestN/service.cs

```diff
@@ -303,5 +304,41 @@ public void Notify(string message)
         {
             Trace.WriteLine(string.Format("Notified about: {0}", message));
         }
+
+        [JsonRpcMethod]
+        public string TestPreProcessor(string inputValue)
+        {
+            return "Success!";
+        }
+
+        [JsonRpcMethod]
+        public string TestPreProcessorThrowsJsonRPCException(string inputValue)
+        {
+            throw new JsonRpcException(-27000, "Just some testing", null);
+        }
+
+        [JsonRpcMethod]
+        public string TestPreProcessorThrowsException(string inputValue)
+        {
+            throw new Exception("TestException");
+        }
+
+        [JsonRpcMethod]
+        public string TestPostProcessor(string inputValue)
+        {
+            return "Success!";
+        }
+
+        [JsonRpcMethod]
+        public string TestPostProcessorThrowsJsonRPCException(string inputValue)
+        {
+            throw new JsonRpcException(-27000, "Just some testing", null);
+        }
+
+        [JsonRpcMethod]
+        public string TestPostProcessorThrowsException(string inputValue)
+        {
+            throw new Exception("TestException");
+        }
```

Comment:
We might as well add some tests for Context.SetException while were at it to both of these. Similar to:
`[JsonRpcMethod("error4")]
        private string throwsException3(string s)
        {
            JsonRpcContext.SetException(new JsonRpcException(-27000, "This exception was thrown using: JsonRpcContext.SetException()", null));
            return null;
        }`